task/tekton-prune/0.1/tekton-prune.yaml

```diff
@@ -0,0 +1,187 @@
+                "--no-execute",
+                type=str,
+                choices=["yes", "true", "false", "no"],
+                help=
+                "Do not delete just print the kubectl command line (to check what will be deleted)."
+            )
+
+            parser.add_argument("--object",
+                                required=True,
+                                type=str,
+                                help="The type of the tekton object to clean.")
+
+            parser.add_argument("--namespace", type=str, help="The target namespace.")
+
+            parser.add_argument(
+                "--max_days",
+                type=str,
+                help="Maximum number of days to keep before deleting the older ones.")
+
+            parser.add_argument("--max_number",
+                                type=str,
+                                help="Maximum number of objects to keep.")
+
+            if len(args) == 0:
+                parser.print_help()
+                sys.exit(1)
+
+            args = parser.parse_args(args)
+
+            if args.max_days and args.max_number:
+                sys.stderr.write(
+                    "--max_number and --max_days are exclusive and cannot be specified at the same time."
+                )
+                sys.exit(1)
+
+            if not args.max_days and not args.max_number:
+                sys.stderr.write(
+                    "neither --max_number or --max_days has been specified.")
+                sys.exit(1)
+
+            if args.max_days and not args.max_days.isdigit():
+                sys.stderr.write(f"{args.max_days} should be a number")
+                sys.exit(1)
+            elif args.max_number and not args.max_number.isdigit():
+                sys.stderr.write(f"{args.max_number} should be a number.")
+                sys.exit(1)
+
+            if args.labels:
+                for label in args.labels.split(","):
+                    if "=" not in label:
+                        sys.stderr.write(
+                            f"label '{label}' does not have an assignment it should be `label=value`"
+                        )
+                        sys.exit(1)
+
+            return args
+
+
+        def main():
+            args = parse_args(sys.argv[1:])
+            labels = [f"-l {k}" for k in args.labels.split(",")] if args.labels else []
+            namespace_str = f"-n {args.namespace}" if args.namespace else ""
+
+            cmdline = f"kubectl {namespace_str} get {args.object} {' '.join(labels)} -o json"
+            output = json.loads(execute(cmdline).stdout.decode())
+
+            def parse_ts(timest):
+                return datetime.datetime.strptime(timest, '%Y-%m-%dT%H:%M:%SZ')
+
+            if args.max_days:
+                results = [
+                    i['metadata']['name'] for i in output['items']
+                    if datetime.datetime.now() >
+                    parse_ts(i['metadata']['creationTimestamp']) +
```

Comment:
Could we use the ['status']['completionTime'] to avoid delete the running object?